src/core/extractor.py

```diff
@@ -52,6 +53,70 @@ async def extract(self, site: WebsiteData, executor: Executor) -> tuple[StarCase
         """
 
 
+async def download_tag_lists(session: ClientSession, urls: Union[str, list[str]], logger: Logger) -> set[str]:
+    """
+    Download tags from a list of urls and combine all downloaded tag lists into set removing all duplicates.
+    :param session: The client to use for the downloads.
+    :param urls: The urls from where to download individual tag lists.
+    :param logger: Logger to be used.
+    :return: The combined set of all tags.
+    """
+
+    def extract_dates(tags: list[str]) -> tuple[Optional[int], Optional[str]]:
+        """
+        Try to extract the expiration and last modification date from a downloaded tag list.
+        :param tags: The individual lines of a downloaded tag list.
+        :return: Tuple holding the expiration (in days?) and last modification date.
+        """
+        expires_expression = re.compile(r"[!#:]\sExpires[:=]\s?(\d+)\s?\w{0,4}")
+        last_modified_expression = re.compile(r"[!#]\sLast modified:\s(\d\d\s\w{3}\s\d{4}\s\d\d:\d\d\s\w{3})")
+        expiration = None
+        last_modification = None
+        for line in tags[0:10]:  # only check first 10 lines
```

Comment:
Please add, that this choice of the first 10 lines is due to the structure of the tag list - at least that is what I remember. Currently, it reads like your personal opinion.